File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& a) {
        unordered_set<int> s;
        int ml=0;
        for(auto e :a)
        {
            s.insert(e);
        }
        for(auto ele:a)
        {
            if(!s.count(ele-1))
            {
                int cn=ele;
                int cc=1;
                while(s.count(cn+1))
                {
                cc=cc+1;
                cn=cn+1;
            }
            ml=max(cc,ml);
            }
        }
        return ml;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length==0)return 0;
        Set<Integer> set = new HashSet<>();
        int longest = 1;
        int length = 1;
        for(int num:nums)
        {
            set.add(num);
        }
        for(int num:set)
        {
            if(!set.contains(num-1))
            {
                length = 1;
                while(set.contains(num+length)) length++;
                longest=Math.max(longest,length);
            }
        }
        return longest;
    }
}